design/object-bucket/object-bucket.md

```diff
@@ -43,115 +51,159 @@ Add a CephObjectBucketClaim API endpoint and control loop to the Rook-Ceph opera
+_As a Kubernetes user, I want to leverage the Kubernetes API to create object store buckets. I expect to get back the bucket endpoint information in a Pod-consumable API resource._
+ 
+//TODO update diagram to reflect API changes  
  
 ![user-actions](./cobc-user.png)
 
-1. The user creates a CephObjectBucketClaim in their namespace.
-1. The operator detects the new ObjectBucketClaim instance.
-    1. The operator derives the object store service via `storageClass.parameters.objectStoreNamespace` + `storageClass.parameters.objectStore`.
-1. If a CephObjectUser for the CephObjectStore does not exist, the operator creates it and waits for the associated Secret.  This will give the operator **bucket** CREATE/GET/DELETE permissions in the object store.  This key is reused later and exists for the lifetime of the object store.
-1. The operator creates the bucket using the service endpoint and CephObjectStoreUser **bucket** access key.  The bucket name will be a concatenation of the  claim's `metadata.namespace` and `metadata.name`.  This protects against bucket naming collisions.
-    1. If the bucket already exists in the store, the operator will instead regenerate the ConfigMap and Secret associated with that bucket.  This allows users the ability to a) reacquire buckets who's claims have been deleted and b) incorporate brownfield buckets created outside of the operator.
-1. The operator creates or updates a ConfigMap in the namespace of the CephObjectBucketClaim with relevant connection data for that bucket.
-    1. The operator uses the radosgw-admin CLI to generate a **user** access key.  This key will be bound to the bucket with an **object** PUT/GET/DELETE ACL.
-    1. The operator binds the **user** key to the bucket with an **object** PUT/GET/DELETE ACL.
-1. The operator creates a Secret in the namespace of the CephObjectBucketClaim with the **user** key pair.
+1. The user creates an OBC in their namespace.
+1. The operator detects the new OBC instance.
+    1. The operator checks the `provisioner` field of the OBC's StorageClass.  If it matches, continue to the next step. Else, end the process here.
+    1. The operator gets the object store service using `storageClass.parameters.objectStoreNamespace` + `storageClass.parameters.objectStore`.
+1. The operator creates a CephObjectUser if it does not exist in order to generate the operator's access key.
+1. The operator creates the bucket using the service endpoint and access key.  The bucket name will be a concatenation of the  OBC's `metadata.namespace` and `metadata.name`.
+    1. If the bucket already exists in the store, the operator will instead regenerate the ConfigMap and Secret associated with that bucket.  This allows users the ability to reacquire buckets who's claims have been deleted.
+1. The operator creates an OB containing metadata about the new bucket and sets it as bound to the OBC.
```

Comment:
"...creates a non-namespaced OB..."

Reply:
I think that may be overly descriptive.  OBs are defined to be non-namespaced in the Glossary section